packages/material-ui/src/Menu/Menu.js

```diff
@@ -43,24 +49,56 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     classes,
     disableAutoFocusItem = false,
     MenuListProps = {},
+    // menuLevel = 1,
     onClose,
+    onEnter,
     onEntering,
+    onEntered,
     open,
     PaperProps = {},
+    setParentLastEnteredItemIndex,
     PopoverClasses,
     transitionDuration = 'auto',
+    nestedMenu = false,
     variant = 'selectedMenu',
     ...other
   } = props;
   const theme = useTheme();
 
+  const [lastEnteredItemIndex, setLastEnteredItemIndex] = useState(null);
+  const [entering, setEntering] = useState(false);
+
+  const atLeastOneNestedMenu = useMemo(() => {
+    return Array.isArray(children)
+      ? nestedMenu ||
+          children.some(
+            (child) =>
+              React.isValidElement(child) &&
+              child.props &&
+              child.props.nestedItems &&
+              child.props.nestedItems.length > 0,
+          )
+      : false;
+  }, [children, nestedMenu]);
+
   const autoFocusItem = autoFocus && !disableAutoFocusItem && open;
 
   const menuListActionsRef = React.useRef(null);
   const contentAnchorRef = React.useRef(null);
 
   const getContentAnchorEl = () => contentAnchorRef.current;
 
+  const handleEnter = (element, isAppearing) => {
+    if (atLeastOneNestedMenu) {
+      setEntering(true);
+      setLastEnteredItemIndex(null);
+    }
+
+    if (onEnter) {
+      onEnter(element, isAppearing);
```

Comment:
do we need to pass isAppearing to all the `onEnter/ing/ed` calls? Do they need anything else other than element?

Reply:
Nevermind, I see this pattern in a few other components